Add Keyboard Lock option into Fullscreen API

xingri opened the pull request:
#### d7997e2d3966356a3d23c894706bac6cccdbe84f
<pre>
Add Keyboard Lock option into Fullscreen API
<a href="https://bugs.webkit.org/show_bug.cgi?id=265923">https://bugs.webkit.org/show_bug.cgi?id=265923</a>

Reviewed by Ryosuke Niwa.

Add option compliant with the fulscreen proposal &lt;<a href="https://github.com/whatwg/fullscreen/issues/231">https://github.com/whatwg/fullscreen/issues/231</a>&gt;.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/FullscreenOptions.h:
* Source/WebCore/dom/FullscreenOptions.idl:

Canonical link: <a href="https://commits.webkit.org/272933@main">https://commits.webkit.org/272933@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/569ae8a047a64b5dac86af8d6cf459434315d7d4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31954 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10653 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33708 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34469 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28950 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32758 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13016 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7889 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28546 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32325 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9009 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28557 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7796 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7975 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28470 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35813 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/27432 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29071 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28924 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34076 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/32023 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8061 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6046 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31935 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9705 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/38464 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7778 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8722 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8166 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8584 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

marcoscaceres commented on Source/WebCore/dom/FullscreenOptions.idl:
```diff
@@ -31,8 +31,17 @@
 };
 
+[
+    Conditional=FULLSCREEN_API
```
We might need a different conditional, just to keep things separated... but it's a start.

nt1m replied:
`Conditional=FULLSCREEN_API` is need either way because the fullscreen API is compiled behind it.

I guess what you meant is a setting or feature flag, in which case this is `EnabledBySetting=`

cdumez commented on Source/WebCore/dom/FullscreenOptions.h:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2022 Apple Inc. All rights reserved.
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
```
2022-2023

cdumez commented on Source/WebCore/dom/FullscreenOptions.idl:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2020 Apple Inc. All rights reserved.
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
```
2020-2023

nt1m commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2789,6 +2789,21 @@ FullScreenEnabled:
 
+FullScreenKeyboardLock:
+  type: bool
+  status: developer
```
Should be testable, developer is for actual developer tools settings.

xingri replied:
@nt1m Thanks for the feedback. Updated.

rniwa reviewed (approved):
Looks okay to me.

xingri committed:
Add Keyboard Lock option into Fullscreen API
https://bugs.webkit.org/show_bug.cgi?id=265923

Reviewed by Ryosuke Niwa.

Add option compliant with the fulscreen proposal <https://github.com/whatwg/fullscreen/issues/231>.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/FullscreenOptions.h:
* Source/WebCore/dom/FullscreenOptions.idl:

Canonical link: https://commits.webkit.org/272933@main
